### ISSUE TITLE
any comparisons with tdmpc2-jax?

### ISSUE THREAD

**Issue by edwhu** [ShaneFlandermeyer/bmpc-jax#3]
Hi, I'd like to get a sense of how this compares to tdmpc2-jax, both in performance and runtime.

**Comment by ShaneFlandermeyer** [comment]
I unfortunately got pulled away from this recently and have not been able to thoroughly profile things. The dog-run reward plot below from [this issue](https://github.com/ShaneFlandermeyer/tdmpc2-jax/issues/13) gives an idea of the performance I've been seeing, where pink=TDMPC2, green=BMPC as described in the paper, and gray=BMPC with actor loss discounting (I am still experimenting with that...see my explanations in the issue for more detail). 

Runtime-wise, it's about 60-70% as fast as TD-MPC2 on my 4090. I think we could speed this up significantly by doing re-planning in an asynchronous thread.


![Image](https://github.com/user-attachments/assets/c3b38fb5-9be6-4023-b4dc-b8316925b19b)

**Comment by wertyuilife2** [comment]
I'm currently running some reproducibility experiments for bmpc-jax. I'll open an issue and share the results in a few days :)

**Comment by ShaneFlandermeyer** [comment]
@wertyuilife2 For profiling/comparison purposes, you should probably use the feature/bmpc branch in tdmpc2-jax. I've made some changes there that I haven't merged back here yet :)

**Comment by wertyuilife2** [comment]
@ShaneFlandermeyer  Damn, I actually already ran a lot of ablation experiments and analyses on the main branch of the bmpc-jax repo. All experiments should be finished by tomorrow, and there are many results to discuss. Maybe once I share my results, you can check if there are any errors caused by version differences (I don’t think there will be many).

**Comment by ShaneFlandermeyer** [comment]
In theory, none of the changes I've made since pushing here should make a difference. Definitely happy to talk about the results when you get them!
